pkg/arvo/sur/chat-store.hoon

```diff
@@ -0,0 +1,47 @@
+|%
+::
+++  serial  @uvH
+::
++$  letter
+  $%  [%text text=cord]
+      [%url url=cord]
+      [%code expression=cord output=(list tank)]
```

Comment:
FWIW, I'm not sure the simplification of message types here is a win. The missing types (from hall) are invites, replies, `@` messages (a la IRC `/me`), long-form/attachments, and app-messages. Of those, only the last can be trivially reconstructed and need not be a separate type. Replies, `@` messages, and long-form all encode user intent, in a way that can be transparently reconstructed. And invites are yet-to-be-built ...

Reply:
Hi Joe. In my opinion, long-form attachments should be written into another service. Similarly, I think threaded data structures and replies should be written as other services. Keep in mind that this is a purposefully opinionated and simple chat data structure, and not the only one that could exist. I think a threaded data structure will be one of the next types of data stores to be written, in fact, but I do not think it will be made to cater to a chat service. Instead, I think it will be written for a microblogging-type UI, which imo threading is well-suited for.

Reply:
At the very least we should just get `@` messages back in. They're a pretty core and simple message type, and get you huge gains in expressiveness.

The reason this isn't moving towards your hellworld of "talk to some friends on Chat, others on Microblog" is that you can, theoretically trivially, write a hook to pull their data into a single store, right? Wouldn't that still be easier if there's more overlap between what the message types allow for?

Reply:
@ messages are unnecessary. They could potentially be added later, but it is not a show-stopping priority for me. I have nothing against them in principle, apart from thinking they are cruft.